docs/helm-chart-api.md

```diff
@@ -0,0 +1,184 @@
+    Load() (*Report, error)
+}
+```
+
+- NewReport: Used to create a new ```Report```.
+  
+- GetContent: Get the report as a string in either json or yaml format. ReportFormat values are defined and available in the report package:
+  - ```JsonReport``` - for json format.
+  - ```YamlReport``` - for yaml format.
+    
+- SetContent: Set the report content from a string, for example a string as returned by ```GetContent```. The format of the report yaml/json will be determined based on the report content.
+  
+- SetUrl: Set the URL of a report, the report being in string format, for example a string as returned by ```GetContent```. The format of the report yaml/json will be determined based on the report content. 
+  
+- Load: Can be used to load a report based on content set using ```SetContent``` or ```SetUrl```. This will be called internally when the report is needed but can be used to check if a report will load without error.
+
+## ReportSummary
+```
+func NewReportSummary() APIReportSummary
+type APIReportSummary interface {
+	SetReport(report *apireport.Report) APIReportSummary
+	GetContent(SummaryType, SummaryFormat) (string, error)
+	SetValues(values map[string]interface{}) APIReportSummary
+}
+
+```
+
+- NewReportSummary: Used to create a ```ReportSummary```.
+  
+- SetReport: Set the report from which the summary should be generated. For example a report as returned by ```report.NewReport```.
+  
+- GetContent: Get the report summary as a string in either json or yaml format. ReportFormat values are defined and availalble in the reportsummary package:
+    - ```JsonReport``` - for json format.
+    - ```YamlReport``` - for yaml format.
+  
+- SetValues: Used to set value flags to tailor content of the report sumary. Can be used to set a profile vendorType and/or version which by default are the values set in the report.
+
+## Checks
+
+```
+func GetChecks() []CheckName
+```
+
+- GetChecks: Used to get an array of ```CheckName``` types. The array content provide value with can be used for ```verifier.EnableChecks``` and ```verifier.UnEnableChecks```:
+    - ```ChartTesting```
+    - ```ContainsTest```
+    - ```ContainsValuesSchema```
+    - ```ContainsValues```
+    - ```HasKubeVersion```
+    - ```HasReadme```
+    - ```HelmLint```
+    - ```ImagesAreCertified```
+    - ```IsHelmV3```
+    - ```NotContainCsiObjects```
+    - ```NotContainsCRDs```
+    - ```RequiredAnnotationsPresent``` 
+
+
+# Example:
+
+Note: Example does not include error checking for clarity.
+
+1. import the api:
+```
+import (
+	"fmt"
+	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/checks"
+	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/report"
+	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/reportsummary"
+	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/verifier"
+)
+```
+
+2. Create a verifier, set a profile vendortype of redhat, un-enable the chart testing check, and run verify:
```

Comment:
Suggestion: This needs a rewrite. For example: "Run the verify command to create a verifier, set a `vendortype` profile as `redhat`, and un-enable the check for chart testing:"

Reply:
Have updated the word, bit not as you suggest as "Run the verify command" is not accurate.